svgdigitizer/pdf.py

```diff
@@ -0,0 +1,162 @@
+r"""
+Module for handling operations related to PDF files.
+"""
+
+import logging
+
+# ********************************************************************
+#  This file is part of svgdigitizer.
+#
+#        Copyright (C) 2025 Johannes Hermann
+#
+#  svgdigitizer is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  svgdigitizer is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with svgdigitizer. If not, see <https://www.gnu.org/licenses/>.
+# ********************************************************************
+import os
+from functools import cached_property
+
+logger = logging.getLogger("svgdigitizer.pdf")
+
+
+class Pdf:
+    "Handles all interactions with the PDF file."
+
+    def __init__(self, pdf_filepath):
+        "Takes the path of a PDF file."
+        self._filepath = pdf_filepath
+
+    @cached_property
+    def doc(self):
+        "Holds the opened PDF file."
+        import pymupdf
+
+        return pymupdf.open(self._filepath)
+
+    def _rename(self, new_name):
+        os.rename(self._filepath, new_name)
+        self._filepath = new_name
+
+    @cached_property
+    def doi(self):
+        """
+        Extract the DOI from the provided PDF. Since in some cases additional pages are prepended to the PDF, the DOI is extracted from either the first or second page.
+        """
+        import re
+
+        doc = self.doc
+
+        for page_num in range(
+            min(doc.page_count, 2)
+        ):  # inspect the first (two) page(s)
+            text = doc.get_page_text(page_num)
+            matches = re.findall(r"10\.\d{4,9}\/[-._;()/:a-zA-Z0-9]+", text)
+            if len(matches) == 0:
+                logger.info(f"No DOI found on page {page_num + 1}. Trying next page.")
+            elif len(matches) > 1:
+                raise ValueError(
+                    f"{len(matches)} DOIs found. Candidates are {matches}. Extraction of DOI failed."
+                )
+            else:
+                logger.info(f"DOI found on page {page_num + 1}.")
+                return matches[0]
+
+        raise ValueError("No DOI found. Extraction of DOI failed.")
+
+    @staticmethod
+    def _download_citation(doi):
```

Comment:
I think this might also be called bibliography or bibliography entry to be in line with our notation in the metadata schema.

Reply:
ok, so here we adhere to the journal, while internally we handle it as bibliography. Fine with me.